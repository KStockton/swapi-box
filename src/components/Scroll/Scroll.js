import React from 'react';
import './Scroll.scss'
import PropTypes from 'prop-types'

const Scroll = ( { filmText } ) => {

return (
  <article className="Scroll-article">
    <p>From a galaxy far far away...</p>
      <div className="filmtext-wrapper-before">
        <div className="filmtext-wrapper-after">
          <h1>{filmText.title}</h1>
          <h4>{filmText.releaseDate}</h4>
          <h5>{filmText.openingCrawl}</h5>
        </div>
      </div>
  </article>
)
}

Scroll.propTypes = {
  filmText: PropTypes.object
}

export default Scroll