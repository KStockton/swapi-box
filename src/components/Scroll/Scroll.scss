.Scroll-article {
  overflow: hidden;
}

.Scroll-galaxy-text {
  font-weight: 800;
  margin: 20% auto;
  color: white;
  opacity: 0;
  z-index: 1;
  animation: fadeTitle 4s ease-out;
  position: fixed;
  left: 36%;
  margin: 9px 7px;
  bottom: 9%;
  font-size: 28px;
}

.filmtext-wrapper-before {
  position: fixed;
  left: 5%;
  bottom: 0%;
  overflow: hidden;
  transform: perspective(250px) rotateX(41deg);
  height: 95%;
  width: 90%;
  &:after {
   position: absolute;
   content: '';
   top: 100%;
    bottom:0;
    left:0;
    right:0;
  }
}

.filmtext-wrapper-after {
  position: absolute;
  top: 100%;
  right: 35%;
  animation: scrollCrawl 50s linear;
  animation-iteration-count: 1;
  color: #ffd900;
  letter-spacing: 8px;
  width: 24rem;
  font-size: 218%;
  text-align: justify;
  text-transform: uppercase;
  font-family: Pathway Gothic One sans-serif;
  background: linear-gradient(0deg, transparent, #0000001a 75%);
  line-height: 105%;
}
