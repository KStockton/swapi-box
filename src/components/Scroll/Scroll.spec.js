import React from 'react';
import { shallow } from 'enzyme';
import Scroll from './Scroll';

const mockProp = {
  title: "Revenge of the Sith",
  releaseDate: "2005-05-19",
  openingCrawl: "War! The Republic is crumbling"
}


describe('Scroll', () => {
  let wrapper;

  beforeEach(() =>{
    wrapper = shallow(<Scroll filmText={mockProp}/>)
  });

  it('should render correctly', () =>{
    expect(wrapper).toMatchSnapshot()
  })
});