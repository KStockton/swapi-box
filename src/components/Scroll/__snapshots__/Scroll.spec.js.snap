// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should render correctly 1`] = `
<article
  className="Scroll-article"
>
  <p />
  <div
    className="filmtext-wrapper-before"
  >
    <div
      className="filmtext-wrapper-after"
    >
      <h1>
        Revenge of the Sith
      </h1>
      <h2>
        Episode: 
      </h2>
      <h4>
        Date: 
        2005-05-19
      </h4>
      <h5>
        War! The Republic is crumbling
      </h5>
    </div>
  </div>
</article>
`;
